src/cpp/src/image_generation/stable_diffusion_pipeline.hpp

```diff
@@ -163,6 +165,19 @@ class StableDiffusionPipeline : public DiffusionPipeline {
         m_vae->compile(vae_device, *updated_properties);
     }
 
+    std::shared_ptr<DiffusionPipeline> clone() override {
+        // m_scheduler is not cloned, but rather set_scheduler exposed for users via GenerationRequest
+        std::shared_ptr<AutoencoderKL> vae = std::make_shared<AutoencoderKL>(m_vae->clone());
+        std::shared_ptr<CLIPTextModel> clip_text_encoder = std::make_shared<CLIPTextModel>(m_clip_text_encoder->clone());
+        std::shared_ptr<UNet2DConditionModel> unet = std::make_shared<UNet2DConditionModel>(m_unet->clone());
+        std::shared_ptr<StableDiffusionPipeline> pipeline = std::make_shared<StableDiffusionPipeline>(
+            m_pipeline_type,
+            *clip_text_encoder,
+            *unet,
+            *vae);
+        return pipeline;
```

Comment:
we probably need to copy generation config as well ?

Reply:
generation config is re-initialized in StableDiffusionPipeline constructor we use here
https://github.com/openvinotoolkit/openvino.genai/blob/f35313b6867d7fecdb904d4e0e1aa685925f3bc3/src/cpp/src/image_generation/stable_diffusion_pipeline.hpp#L130

Reply:
That's unfortunate `StableDiffusionPipeline` needs file system access even though everything is already in memory.
Would adding `StableDiffusionPipeline() = default` and cloning every member solve that?

Reply:
Even if it would solve that, we still need file system access at runtime due to fact that Scheduler is re-created using saved `root_dir`:

```
pipeline->m_root_dir = m_root_dir;
pipeline->set_scheduler(
  Scheduler::from_config(m_root_dir / "scheduler/scheduler_config.json"));
```

Do you see any way around that? Adding clone to all kinds of schedulers seems like overkill if we can create one from config file

Reply:
@likholat, does `Scheduler` change it's state during pipeline usage? If not we, the pipeline could store a shared pointer

Reply:
@Wovchena yes it changes the state, for example in `set_timesteps` function. I don't think we can reuse the scheduler

Reply:
Is using saved root_dir a no-go?

Reply:
If the json changes between clone() calls the instances won't be identical. So the final solution can't read the file system.

At the same time this scenario is unlikely. Can you try one of the `Scheduler` with clone() just to ensure the problem can be resolved? After that you can come back to json reading as a temporary solution if @likholat is OK with this plan.

Reply:
I think reading from json is fine at this point

Reply:
In separate task: find a way to clone scheduler without reading from disk